[Term Entry] C++ vector .size() (#4994)
Codecademy/docs: content/cpp/concepts/vectors/terms/size/size.md

---
Title: '.size()'
Description: 'Returns the number of elements in the vector.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Vectors'
  - 'Arrays'
  - 'Data Structures'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`.size()`** method returns the number of elements in the vector.
> `.size()` is there to follow the consistency of other standard library containers (e.g., [maps](https://www.codecademy.com/resources/docs/cpp/maps) and [strings](https://www.codecademy.com/resources/docs/cpp/strings)).

Reviewer: ```suggestion
The **`.size()`** method returns the number of elements in the vector. It follows the consistency of other standard library containers, such as [maps](https://www.codecademy.com/resources/docs/cpp/maps) and [strings](https://www.codecademy.com/resources/docs/cpp/strings).
```

Reviewer: Reframed for better clarity

## Syntax

```pseudo
vector.size();
```

> The `vector` must be defined with `std::vector` prior to using with the `.size()` method.

Reviewer: ```suggestion
The `vector` must be defined using `std::vector` before the `.size()` method can be used.
```

> ## Codebyte Example

Reviewer: ```suggestion
## Example
```

Reviewer: For this issue, an Example block is needed

In the example below, `.size()` is called on the `numbers` vector:

> ```codebyte/cpp

Reviewer: ```suggestion
```cpp
```

Reviewer: In the example block, we use the ```cpp block. If it was a codebyte block, then we use the ```codebyte/cpp

#include <iostream>
#include <vector>

int main() {
    // Declaring a vector with 4 integers
    std::vector<int> numbers = {1, 2, 3, 4};

    // Print out vector size
    std::cout << numbers.size();
}

> ```

Reviewer: Please add an output of the above code block below as follows: (use the ```shell``` block)

The output of the above code is:

```shell
4
```
